File: LeetCode/Easy/13.RomanToInteger/Python/RomanToInteger.py
def main():
    s = input("Enter the Roman Numeral : ")

    result = RomanToInteger(s.upper())

    print(f"The Decimal equivalent for the entered roman numeral({s}) is {result}")

def RomanToInteger(str):
    result = 0

    RtoI = {
        'I' : 1,
        'V' : 5,
        'X' :10,
        'L' : 50,
        'C' : 100,
        'D' : 500,
        'M' : 1000
    }

    for i in range(len(str) - 1):
        if RtoI[str[i]] < RtoI[str[i + 1]]:
            result = result - RtoI[str[i]]
        else:
            result = result + RtoI[str[i]]

    return result + RtoI[str[len(str) - 1]]

if __name__ == '__main__':
    main()

File: LeetCode/Easy/13.RomanToInteger/Java/RomanToInteger.java
import java.util.Scanner;
import java.util.HashMap;

public class RomanToInteger {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the Roman Numeral : ");
        String s = scanner.nextLine();

        int result = RIConverter(s.toUpperCase());

        System.out.printf("The Decimal equivalent of the entered Roman Numeral(%s) is : %d\n",s,result);

        scanner.close();
    }

    public static int RIConverter(String s) {
        int result = 0;

        HashMap<Character,Integer> RtoI = new HashMap<>();
        RtoI.put('I',1);
        RtoI.put('V',5);
        RtoI.put('X',10);
        RtoI.put('L',50);
        RtoI.put('C',100);
        RtoI.put('D',500);
        RtoI.put('M',1000);

        for (int i = 0;i < s.length() - 1;i++) {
            if (RtoI.get(s.charAt(i)) < RtoI.get(s.charAt(i + 1))) {
                result = result - RtoI.get(s.charAt(i));
            } else {
                result = result + RtoI.get(s.charAt(i));
            }
        }

        return result + RtoI.get(s.charAt(s.length() - 1));
    }
}